### PR TITLE
Summary page show licenses

### PR DESCRIPTION
**Pull request by Tanya0609**
Task:https://microsoftgarage.visualstudio.com/Intern%20GitHub/_sprints/taskboard/WTS%20Team/Intern%20GitHub/Y19-VS/06
Task:https://microsoftgarage.visualstudio.com/Intern%20GitHub/_sprints/taskboard/WTS%20Team/Intern%20GitHub/Y19-VS/07

This change is to make current summary page (middle panel) as the expected screenshot. Changes included in my code:
- update summary page into show less information
- add view licenses modal
- able to resize the modals based on modal type ( so far only _post gen modal_ and _view licenses modal_ have width 30%, the others have width 40%)

**How to test:**
1. generate any project type with any page
2. at summary page, clicks on **View Licenses** button, it opens a modal and contains all the corresponding licenses
3. at summary page, clicks on **Give Feedback** button, it should redirect you to GitHub Issue page

**Expected from my PR:**
licensesmodal

**New Summary Page from Designer:** ( the background image in this design is from the internet)
summary page

### CODE REVIEW

File: src/client/src/components/Modal/index.tsx [hunk 1]
```diff
@@ -16,7 +17,19 @@ import Modal from "react-modal";
...
-const getCustomStyles = (isPostGenModalOpen: boolean) => {
+const getCustomStyles = (MODAL_TYPE: string | undefined) => {
+  // default width
+  let CUSTOM_WIDTH = "40%";
+
+  // depends on modal type, customize width
+  if (
+    MODAL_TYPE &&
+    (MODAL_TYPE == MODAL_TYPES.POST_GEN_MODAL ||
+      MODAL_TYPE == MODAL_TYPES.VIEW_LICENSES_MODAL)
```
Comment (streetcoder123) [inline src/client/src/components/Modal/index.tsx]: We should have `===` instead of `==`

Comment (Tanya0609) [reply src/client/src/components/Modal/index.tsx]: good catch, updated PR

File: src/client/src/components/Modal/index.tsx [hunk 2]
```diff
@@ -16,7 +17,19 @@ import Modal from "react-modal";
  * Custom styling guidance:
  * https://reactcommunity.org/react-modal/styles/
  */
-const getCustomStyles = (isPostGenModalOpen: boolean) => {
+const getCustomStyles = (MODAL_TYPE: string | undefined) => {
```
Comment (jsondoo) [inline src/client/src/components/Modal/index.tsx]: should we use `ModalType` here instead of string?

File: src/client/src/components/Modal/index.tsx [hunk 3]
```diff
@@ -49,7 +62,8 @@ interface IProps {
 }
 
 const asModal = <P extends object>(
-  WrappedComponent: React.ComponentType<P>
+  WrappedComponent: React.ComponentType<P>,
+  MODAL_TYPE?: string
```
Comment (jsondoo) [inline src/client/src/components/Modal/index.tsx]: also here

File: src/client/src/components/Modal/index.tsx [hunk 4]
```diff
@@ -49,7 +62,8 @@ interface IProps {
...
 ) => {
   return class extends React.Component<P & IProps> {
     static defaultProps = {
```
Comment (jsondoo) [inline src/client/src/components/Modal/index.tsx:68]: I think we can remove `isPostGenModalOpen` from default props now

File: src/client/src/containers/PostGenerationModal/index.tsx [hunk 5]
```diff
@@ -231,5 +232,5 @@ export default withRouter(
   connect(
     mapStateToProps,
```
Comment (jsondoo) [inline src/client/src/containers/PostGenerationModal/index.tsx:231]: Also remove `isPostGenModalOpen` prop from this component

File: src/client/src/components/Modal/index.tsx (hunk 3 above)
Comment (Tanya0609) [reply src/client/src/components/Modal/index.tsx]: Good Catch! updated in PR

File: src/client/src/components/Modal/index.tsx (hunk 4 above)
Comment (Tanya0609) [reply src/client/src/components/Modal/index.tsx:68]: updated in the PR

**Review by streetcoder123 — COMMENTED** [review]
@Tanya0609 When I click on `Give Feedback`, it redirects me to the GitHub issues page, which is good. However, I think we should open the GitHub issues page in a new tab so that the wizard remains there in a browser tab. You will just need to set the target for this.
`target="_blank"`

**Comment by Tanya0609** [comment]
@njindal239 This won't happen if a regular user clicks the link through the extension, however, this will improve developers' experience. I will update my PR to include this change

File: src/client/src/containers/ReviewAndGenerate/index.tsx [hunk 6]
```diff
@@ -1,120 +1,89 @@
...
-      />
-      <div className={styles.selectionContainer}>
-        <div className={styles.selectionTitle}>
-          {intl.formatMessage(messages.pages)}
+        <div className={styles.buttonContainer}>
+          <button
+            className={classnames(buttonStyles.buttonDark, styles.leftButton)}
+            onClick={openViewLicensesModal}
+          >
+            {formatMessage(messages.viewLicenses)}
```
Comment (jsondoo) [inline src/client/src/containers/ReviewAndGenerate/index.tsx:58]: NIT: view licenses button is not tabbable but u could do it in a separate PR

**Review by streetcoder123 — APPROVED** [review]

**Review by jsondoo — APPROVED** [review]

**Review by jsondoo — APPROVED** [review]
Talked to @Tanya0609 and she will address tab issues in a separate PR

File: src/client/src/containers/ViewLicensesModal/index.tsx [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+type Props = IStateProps & InjectedIntlProps & IDispatchProps;
+
+const messages = defineMessages({
+  closeModalLabel: {
+    id: "viewLicensesModal.closeModalLabel",
+    defaultMessage: "Close"
+  }
+});
+
+const ViewLicensesModal = ({ intl, closeModal }: Props) => {
```
Comment (piotrmark) [inline src/client/src/containers/ViewLicensesModal/index.tsx:34]: That's a good candidate for unit tests ;)

**Review by piotrmark — APPROVED** [review]

File: src/client/src/containers/ViewLicensesModal/index.tsx (hunk 7 above)
Comment (Tanya0609) [reply src/client/src/containers/ViewLicensesModal/index.tsx:34]: Okay, I will log it first 👍
